tests/main.test.ts

```diff
@@ -1,62 +1,226 @@
 import { drop } from "@mswjs/data";
 import { TransformDecodeError, Value } from "@sinclair/typebox/value";
-import program from "../src/parser/payload";
-import { run } from "../src/run";
+import { runPlugin } from "../src/run";
 import { userActivityWatcherSettingsSchema } from "../src/types/plugin-inputs";
-import { db as mockDb } from "./__mocks__/db";
-import dbSeed from "./__mocks__/db-seed.json";
+import { db } from "./__mocks__/db";
 import { server } from "./__mocks__/node";
 import cfg from "./__mocks__/results/valid-configuration.json";
+import { expect, describe, beforeAll, beforeEach, afterAll, afterEach } from "@jest/globals";
+import dotenv from "dotenv";
+import { Logs } from "@ubiquity-dao/ubiquibot-logger";
+import { Context } from "../src/types/context";
+import mockUsers from "./__mocks__/mock-users";
+import { botAssignmentComment, getIssueHtmlUrl, getIssueUrl, noAssignmentCommentFor, STRINGS, updatingRemindersFor } from "./__mocks__/strings";
+import { createComment, createIssue, createRepo, ONE_DAY } from "./__mocks__/helpers";
+import { collectLinkedPullRequests } from "../src/handlers/collect-linked-pulls";
 
-beforeAll(() => server.listen());
-afterEach(() => server.resetHandlers());
-afterAll(() => server.close());
+dotenv.config();
+const octokit = jest.requireActual("@octokit/rest");
 
-jest.mock("../src/parser/payload", () => {
-  // Require is needed because mock cannot access elements out of scope
-  const cfg = require("./__mocks__/results/valid-configuration.json");
-  return {
-    stateId: 1,
-    eventName: "issues.assigned",
-    authToken: process.env.GITHUB_TOKEN,
-    ref: "",
-    eventPayload: {
-      issue: { html_url: "https://github.com/ubiquibot/user-activity-watcher/issues/1", number: 1, assignees: [{ login: "ubiquibot" }] },
-      repository: {
-        owner: {
-          login: "ubiquibot",
-        },
-        name: "user-activity-watcher",
-      },
-    },
-    settings: cfg,
-  };
+beforeAll(() => {
+  server.listen();
+});
+afterEach(() => {
+  drop(db);
+  server.resetHandlers();
 });
+afterAll(() => server.close());
 
-describe("Run tests", () => {
-  beforeAll(() => {
-    drop(mockDb);
-    for (const item of dbSeed.issues) {
-      mockDb.issues.create(item);
-    }
+describe("User start/stop", () => {
+  beforeEach(async () => {
+    await setupTests();
   });
 
   it("Should parse thresholds", async () => {
     const settings = Value.Decode(userActivityWatcherSettingsSchema, Value.Default(userActivityWatcherSettingsSchema, cfg));
-    expect(settings).toEqual({ warning: 302400000, disqualification: 604800000, watch: { optOut: ["private-repo"] } });
+    expect(settings).toEqual({ warning: 302400000, disqualification: 604800000, watch: { optOut: [STRINGS.PRIVATE_REPO_NAME] } });
     expect(() =>
       Value.Decode(
         userActivityWatcherSettingsSchema,
         Value.Default(userActivityWatcherSettingsSchema, {
           warning: "12 foobars",
```

Comment:
Maybe this default value should be changed to something meaningful

Reply:
Which default value do you mean sorry? `optOut`? This was one of the original tests the plugin had

Reply:
I just understood that it is testing weird values on purpose, didn't have the full context of the test, apologies.